frame/liquidations/src/lib.rs

```diff
@@ -58,47 +66,182 @@ pub mod pallet {
+			StrategyIndex::<T>::mutate(|x| {
+				*x = x.next();
+				*x
+			})
+		}
+	}
+
+	#[derive(Clone, Debug, Encode, Decode, TypeInfo, PartialEq)]
+	pub enum LiquidationStrategyConfiguration<ParachainId> {
+		DutchAuction(TimeReleaseFunction),
+		UniswapV2 { slippage: Perquintill },
+		XcmDex { parachain_id: ParachainId },
+		/* Building fully decoupled flow is described bellow. Will avoid that for now.
+		 * ```plantuml
+		 * `solves question - how pallet can invoke list of other pallets with different configuration types
+		 * `so yet sharing some liquidation part and tracing liquidation id
+		 * dutch_auction_strategy -> liquidation : Create new strategy id
+		 * dutch_auction_strategy -> liquidation : Add Self Dispatchable call (baked with strategyid)
+		 * liquidation -> liquidation: Add liquidation order
+		 * liquidation -> liquidation: Get Dispatchable by Strategyid
+		 * liquidation --> dutch_auction_strategy: Invoke Dispatchable
+		 * dutch_auction_strategy -> dutch_auction_strategy: Get liquidation configuration by id previosly baked into call
+		 * dutch_auction_strategy --> liquidation: Pop next order
+		 * dutch_auction_strategy -> dutch_auction_strategy: Start liqudaiton
+		 * ```
+		 *Dynamic { liquidate: Dispatch, minimum_price: Balance }, */
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			let index = Pallet::<T>::create_strategy_id();
+			DefaultStrategyIndex::<T>::set(index);
+			let linear_ten_minutes = LiquidationStrategyConfiguration::DutchAuction(
+				TimeReleaseFunction::LinearDecrease(LinearDecrease { total: 10 * 60 }),
+			);
+			Strategies::<T>::insert(index, linear_ten_minutes);
+
+			let index = Pallet::<T>::create_strategy_id();
+			let exponential =
+				StairstepExponentialDecrease { step: 10, cut: Permill::from_rational(95_u32, 100) };
+			let exponential = LiquidationStrategyConfiguration::DutchAuction(
+				TimeReleaseFunction::StairstepExponentialDecrease(exponential),
+			);
+			Strategies::<T>::insert(index, exponential);
+		}
+	}
+
+	impl<T: Config> Liquidation for Pallet<T> {
+		type LiquidationStrategyId = T::LiquidationStrategyId;
 
-		type GroupId = T::GroupId;
+		type OrderId = T::OrderId;
 
 		fn liquidate(
-			_source_account: &Self::AccountId,
-			_source_asset_id: Self::AssetId,
-			_source_asset_price: PriceStructure<Self::GroupId, Self::Balance>,
-			_target_asset_id: Self::AssetId,
-			_target_account: &Self::AccountId,
-			_total_amount: Self::Balance,
-		) -> Result<Self::LiquidationId, DispatchError> {
-			Self::deposit_event(Event::<T>::PositionWasSentToLiquidation {});
-			Err(DispatchError::Other("todo"))
+			from_to: &Self::AccountId,
+			order: Sell<Self::MayBeAssetId, Self::Balance>,
+			configuration: Vec<Self::LiquidationStrategyId>,
+		) -> Result<T::OrderId, DispatchError> {
+			if configuration.is_empty() {
+				let configuration = Strategies::<T>::get(DefaultStrategyIndex::<T>::get())
```

Comment:
To avoid having an `expect`, we could perhaps decouple the storage `Strategies` with a second `DefaultStrategy` storage?